code/src/components/TaskList.js

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import tasks from 'reducers/tasks'
+import styled from 'styled-components'
+import { Button } from 'components/GlobalStyles'
+import { FaTrash } from 'react-icons/fa';
+import moment from 'moment';
+
+const TaskList = () => {
+  const taskList = useSelector((store) => store.tasks.items);
+  const dispatch = useDispatch();
+
+  const onDeleteItemClick = (todoIndex) => {
+    dispatch(tasks.actions.deleteItem(todoIndex));
+  };
+
+  const onCompleteToggle = (id) => {
+    dispatch(tasks.actions.toggleItem(id))
+  };
+
+  return (
+    <StyledList>
+      {taskList.map((singleTask, index) => {
+        return (
+          <ListWrapper>
+            <div>
+              <input
+                className="checkbox"
+                name="tasks"
+                type="checkbox"
+                checked={singleTask.complete}
+                onChange={() => onCompleteToggle(singleTask.id)} />
```

Comment:
Oh no, this doesn't seem to work :( When I press the check button it doesn't change state.